modules/apps/frontend-data-set/frontend-data-set-api/src/main/java/com/liferay/frontend/data/set/filter/BaseAutocompleteFDSFilter.java

```diff
@@ -26,8 +26,8 @@ public abstract class BaseAutocompleteFDSFilter implements FDSFilter {
 
 	public abstract String getAPIURL();
 
-	public String getEntityFieldType() {
-		return "collection";
+	public String getFieldIndexingType() {
+		return "notIndexed";
 	}
```

Comment:
> FieldIndexingType.NOT_INDEXED = "notIndexed";  // lambda operator

I'm just confusing about "notIndexed" fieldIndexingType for lambda operator,

"not indexed" means that the field content is not indexed in ES, so it couln't be filtered.

Reply:
@MarinhoFeliphe Going through all of this with fresh eyes, after PTO, I believe I have jumped to the wrong conclusions. You are right, lambda operator doesn't have anything to do with indexing. Hopefully I'm understanding it correctly this time:

Indexing of the filed determines two branches:
1. If the field is indexed, we can use the `filter` parameter, using comparison operators, including lambdas
2. If the field is not indexed, we must use the `search` parameter.

With this in mind, we don't need to add API for indexing type. We may need to some day, when we decide to support non-indexed fields, but that's outside of scope of this task. I'm sorry to have led you to this unnecessary refactoring :disappointed: 

I also finally understand what you mean my collection entity type. [This example helped](https://docs.microsoft.com/en-us/odata/concepts/queryoptions-usage#lambda-operators). For some reason, I imagined the collection we were discussing to be the set of field values of _all_ items. But no, it is the value of a field of a _single_ item, that happens to be a collection. `any` / `all` of the lambda operator suddenly make sense! Also, our current API, `getEntityFieldType`, does make sense, as the `collection` entity field type **is** the true determining factor for odata format.

Finally, my main question is: what's wrong with the current API in `master`? We don't need to change `getEntityFieldType` to `getOperator`, as was the original PR. We don't need to change `getEntityFieldType` to `getFieldIndexingType`, as in current PR. If this is only about the support for `in` operator, we are [already supporting it](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/AutocompleteFilter.js#L82-L92). The only thing I would improve is to [tweak these conditions](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/AutocompleteFilter.js#L115). I believe we even have a bug in the case of single selection of collection type entity.

Can `in` be combined with `lambda`? Is this valid:
```js
?filter=countriesVisited/any(x:(x in ('brasil', 'spain')))
```

If so, we could make something like
```js
if (itemsValues.length > 1) {
    // build condition with in
} else {
    // build condition with eq
}

if (entityFieldType === 'collection'){
    // return condition with `any` lambda wrapper
}

return condition;
```

Reply:
Hi @markocikos welcome back!, I hope you had a greate PTO :smiley: ,

> Indexing of the field determines two possible odata formats:
> 
> If the field is indexed, we can use the filter parameter, using comparison operators, including lambdas
> If the field is not indexed, we must use the search parameter.

Just adjusting,

If the field is indexed as keyword, we can use the `filter` parameter, using comparison operators, including lambdas
If the field is indexed as text, we must use the `search` parameter.
If the field is not indexed, we can't search

> what's wrong with the current API in master?

In my opnion the best way is using the `getEntityFieldType` but Daniel Sanz opened [this ticket](https://issues.liferay.com/browse/LPS-154194) and ask me to take a look,

I didn't know how to make this API more agnostic than using `getEntityFieldType` so I opened this pull request to collaborate with your team on how I could do this.

> Can in be combined with lambda?

This combination isn't yet supported by the headless team, I tried this yesterday but no success.

> I believe we even have a bug in the case of single selection of collection type entity.

Yes, we don't provide that combination, but I can provide that combination in this pull request if you want, I just need to deliver two tasks this week, I can work on it as soon as possible.

Reply:
@MarinhoFeliphe Thanks for clarifications!

One thing that is weird in this API is that we are defining the entity field type in _filter_. Ideally, we should be able to get the field type from item entries. For example, in commerce products there is a filter on product categories, which is a collection (makes sense, a product can have multiple categories). In `items`, taken from return data JSON, `categories` is an array.

A weird case that goes against this logic is `status`. In `items` it is an object, yet it is an collection 'under the hood', and requires lambda format. IIRC, you said that this is probably a mistake. I am not sure if this is something we need to or want to support.

Maybe it makes sense to:
1. Use type from `items` as default. If it's an array, use lambda, else use `in` (`eq` if one selected).
2. Use `entityFieldType` as a way to override above behavior. Set it to `null` by default.

What do you think?

Sidenote: Eventually, we can drop `entityFieldType` for API to set a custom filter `odataString` through `propsTransformer`. That can cover oddities like `status` and different date formats.

Reply:
@markocikos exactly!

`status` field mapped as collection is a mistake mapping, `categories` and `tags` are examples of collection type as you said.

This approach looks good to me!

Summarize, are we going to keep the `entityFieldType` and do I need to add another prop called `itemType` to know if the item is an array?

let me know if I missing something.

Thank you!

Reply:
@MarinhoFeliphe 

> do I need to add another prop called itemType in BaseAutocompleteFDSFilter to know if the item is an array, string, number, ...?

I don't think we need to. Value of item will be of a certain type, and it doesn't make sense to have a prop that claims otherwise. For example, let's say that the value is string, and `itemType` is `array`. This will only lead to a runtime error, as we would handle string as an array. IMO what we should do is check for value type and handle it accordingly. In fact, we are already doing this, [adding quotes if the value is a string](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/AutocompleteFilter.js#L96). In addition to string check, we would now add an array check.